File: LC1309.cpp
class Solution {
public:
    string freqAlphabets(string s) {
        int i = 0, l = s.length();
        string res;
        while(i < l){
            char c;
            if(i+2 < l && s[i+2] == '#'){
                c = 'a' + (s[i] - '0')*10 + (s[i+1] - '0') - 1;
                res.push_back(c);
                i += 3;
            }
            else{
                c = 'a' + (s[i] - '0') -1;
                res.push_back(c);
                i++;
            }
        }
        return res;
    }
};

File: LC1309.py
class Solution:
    def freqAlphabets(self, s: str) -> str:
        i = 0
        l = len(s)
        res = ''

        while i < l:

            if i+2 < l and s[i+2] == '#':
                c = ord('a') + int(s[i])*10 + int(s[i+1]) - 1
                i += 3
            else:
                c = ord('a') + int(s[i]) - 1
                i += 1
            
            res += chr(c)


        return res
